Provide docker for deployment

shenkeyao opened the pull request:
Closes https://app.asana.com/1/1208976916964769/project/1209392461754458/task/1210286484253196?focus=true.

### This PR:
* Redoes https://github.com/EspressoSystems/optimism-espresso-integration/pull/151, which was based on `celo-integration`.
* Adds `docker-compose` for deployment.
* Updates the `Dockerfile` for `op-deployer`.

### How to test this PR: 
* `docker compose up --build -d && docker compose logs -f` to test the Docker Compose file.
  * To filter logs for a specific service, e.g., Caff node, append `| grep caff-node-1` to the command above.
  * To test one service specifically (without starting all other services except its dependencies), e.g., Caff node, run `docker compose up caff-node`. 
* `docker compose down` to shut down all services.
  * Don't add `-v` to this command unless there's an issue that requires volume removals. With `-v`, it will regenerate the hashes and cause a hash mismatch error. If you accidentally did this, the easiest fix would be to replace the hash in `rollup_devnet.json` with the updated one.
  
### Notes:
* The Docker Compose file and the genesis files may contain unnecessary configurations. I'd like to make sure everything works before cleaning them up.
* For the same reason, this branch hasn't been synced with the latest main branch yet.
* I don't see concerning errors in the logs, but please let me know if anyone notices anything worth double-checking.

shenkeyao committed:
Add docker compose and update dockerfile

shenkeyao committed:
Add env, remove ports

shenkeyao committed:
Remove optional script

shenkeyao committed:
Fix build

shenkeyao committed:
Merge branch 'celo-integration-rebase-13' into keyao/docker-for-deploy

shenkeyao committed:
Fix in progress

shenkeyao committed:
Merge branch 'celo-integration-rebase-13' into keyao/docker-for-deploy

shenkeyao committed:
Fix services

dailinsubjam commented:
I gave it a try, and here’s what I observed:
- The logs from `espresso-dev-node-1` are too often and tend to drown out logs from other components.
- I saw similar error messages across multiple components, like this:

``` shell
op-node-verifier-1   | t=2025-06-02T17:00:39+0000 lvl=warn msg="failed to dial address, but may connect later" addr=http://l1:8545
...
op-node-verifier-1 exited with code 1
```
This happened not just with `op-node-verifier-1`, but also with `op-node-sequencer-1`, `op-batcher-1` etc. Maybe we need to tweak some settings for docker network.

shenkeyao committed:
Fix l1 connection

shenkeyao commented:
@dailinsubjam This should be fixed now! I still got this error on initiation, but it started to work after a while, and I didn't see it anymore for 20 min.

shenkeyao commented:
Reviewers: I'll move the Docker Compose file to the `espresso` directory. This won't change the file content, except for some paths, so it does not block the review.

dailinsubjam commented:
@shenkeyao I still get the same errors, and they'll end with 
```
lvl=crit msg="Application failed" message="failed to setup: failed to dial L1 RPC: operation failed permanently after 30 attempts: address unavailable (http://l1:8545)"
```

shenkeyao commented:
@dailinsubjam did you run `docker compose up --build -d && docker compose logs -f` on the latest commit? It's weird that I don't get this `operation failed permanently after 30 attempts` error locally.

philippecamacho commented on ops/docker/deployment-utils/Dockerfile:
```diff
@@ -33,3 +33,10 @@ COPY --from=base /root/.foundry/bin/forge /usr/local/bin/forge
 COPY --from=go-base /go/bin/dasel /usr/local/bin/dasel
+
+# Install geth
+RUN curl -L https://gethstore.blob.core.windows.net/builds/geth-linux-amd64-1.15.11-36b2371c.tar.gz \
```
Could we check the hash of the file we download?

shenkeyao replied:
Good idea, done!

philippecamacho commented on config/op-node/rollup_devnet.json:
```diff
@@ -0,0 +1,27 @@
+{
```
How is this file generated?

shenkeyao replied:
It's mostly copy-pasted from [901-rollup.json](https://github.com/EspressoSystems/optimism-espresso-integration/blob/1c9f7087dcec889c33216f1e698919d054f94ba8/op-program/chainconfig/test/configs/901-rollup.json), with some unused info skipped.

philippecamacho commented on config/jwt.txt:
```diff
@@ -0,0 +1 @@
+0x5f7af7b44f9ca407319c2a133c26f782c89ec84e8112e19c5ca83f2c21a754b8
```
How is this address generated?

shenkeyao replied:
It's an auto-generated JWT secret required by OP Stack. I just removed it since I realized that running the Docker file would actually update this secret each time.

shenkeyao committed:
Add hash check, remove auto generated file

shenkeyao committed:
Move to espresso dir

philippecamacho commented:
I have some errors too:
```
> $ docker compose up --build -d && docker compose logs -f | grep caff-node-1

caff-node-1          | t=2025-06-04T23:53:23+0000 lvl=info msg="Not opted in to ProtocolVersions signal loading, disabling ProtocolVersions contract now."
caff-node-1          | t=2025-06-04T23:53:23+0000 lvl=info msg="Using default bootnodes, none provided."
caff-node-1          | t=2025-06-04T23:53:23+0000 lvl=info msg="Rollup Config" l2_chain_id=901 l2_network="unknown L2" l1_chain_id=900 l1_network="unknown L1" l2_start_time=1728358574 l2_block_hash=0x2a92ff72dad302d39fa80ef81522f0ccb27dc903255b618dfc4feddb22a8f80d l2_block_number=0 l1_block_hash=0xf39446e09aeca67452545d06a6e6a6a11184575ecf421f9306cf3602febf93ba l1_block_number=1 regolith_time="(not configured)" canyon_time="(not configured)" delta_time="(not configured)" ecotone_time="(not configured)" fjord_time="(not configured)" granite_time="(not configured)" holocene_time="(not configured)" isthmus_time="(not configured)" jovian_time="(not configured)" interop_time="(not configured)" cel2_time="(not configured)"
caff-node-1          | t=2025-06-04T23:53:23+0000 lvl=info msg="Initializing rollup node" version=v0.0.0
caff-node-1          | t=2025-06-04T23:53:23+0000 lvl=error msg="Error initializing the rollup node" err="failed to init L1: failed to validate the L1 config: failed to get L1 chain ID: 403 Forbidden: invalid host specified\n"
caff-node-1          | t=2025-06-04T23:53:23+0000 lvl=crit msg="Application failed" message="failed to setup: unable to create the rollup node: failed to init L1: failed to validate the L1 config: failed to get L1 chain ID: 403 Forbidden: invalid host specified\n"
caff-node-1 exited with code 1
```

shenkeyao commented:
> I have some errors too:
> 
> ```
> > $ docker compose up --build -d && docker compose logs -f | grep caff-node-1
> 
> caff-node-1          | t=2025-06-04T23:53:23+0000 lvl=info msg="Not opted in to ProtocolVersions signal loading, disabling ProtocolVersions contract now."
> caff-node-1          | t=2025-06-04T23:53:23+0000 lvl=info msg="Using default bootnodes, none provided."
> caff-node-1          | t=2025-06-04T23:53:23+0000 lvl=info msg="Rollup Config" l2_chain_id=901 l2_network="unknown L2" l1_chain_id=900 l1_network="unknown L1" l2_start_time=1728358574 l2_block_hash=0x2a92ff72dad302d39fa80ef81522f0ccb27dc903255b618dfc4feddb22a8f80d l2_block_number=0 l1_block_hash=0xf39446e09aeca67452545d06a6e6a6a11184575ecf421f9306cf3602febf93ba l1_block_number=1 regolith_time="(not configured)" canyon_time="(not configured)" delta_time="(not configured)" ecotone_time="(not configured)" fjord_time="(not configured)" granite_time="(not configured)" holocene_time="(not configured)" isthmus_time="(not configured)" jovian_time="(not configured)" interop_time="(not configured)" cel2_time="(not configured)"
> caff-node-1          | t=2025-06-04T23:53:23+0000 lvl=info msg="Initializing rollup node" version=v0.0.0
> caff-node-1          | t=2025-06-04T23:53:23+0000 lvl=error msg="Error initializing the rollup node" err="failed to init L1: failed to validate the L1 config: failed to get L1 chain ID: 403 Forbidden: invalid host specified\n"
> caff-node-1          | t=2025-06-04T23:53:23+0000 lvl=crit msg="Application failed" message="failed to setup: unable to create the rollup node: failed to init L1: failed to validate the L1 config: failed to get L1 chain ID: 403 Forbidden: invalid host specified\n"
> caff-node-1 exited with code 1
> ```

@philippecamacho Weird that I didn't get this error. Probably because you specifically checked the Caff node--I'll rerun it!

shenkeyao committed:
Add genesis file

shenkeyao commented:
@philippecamacho Could you try `docker compose up --build -d && docker compose logs -f | grep caff-node-1` again and see if it works? Even better is to run `docker compose down -v && docker volume prune -f` first, but this may be slow.

For context, I sent an update which _theoretically_ should fix the error, but Docker Compose doesn't seem to recognize the new `l1_genesis` file I added. It previously compiled with the new file and failed with an error, but after I fixed the error, it never used the file again and always failed with the incorrect chain ID. I tried clearing and restarting everything, but it still failed.

So I was wondering whether you'd see the same chain ID error, which would confirm whether it's a machine issue.

philippecamacho commented:
I get the same error:
```
> git log
commit 8dcb4cc540b31236575be2a83353d7c3ce82ac50 (HEAD -> keyao/docker-for-deploy, origin/keyao/docker-for-deploy)
Author: Keyao Shen <shenkeyao@gmail.com>
Date:   Wed Jun 4 22:14:20 2025 -0700

    Add genesis file

>  docker compose up --build -d && docker compose logs -f | grep caff-node-1

caff-node-1          | t=2025-06-05T11:50:25+0000 lvl=info msg="Not opted in to ProtocolVersions signal loading, disabling ProtocolVersions contract now."
caff-node-1          | t=2025-06-05T11:50:25+0000 lvl=info msg="Using default bootnodes, none provided."
caff-node-1          | t=2025-06-05T11:50:25+0000 lvl=info msg="Rollup Config" l2_chain_id=901 l2_network="unknown L2" l1_chain_id=900 l1_network="unknown L1" l2_start_time=1728358574 l2_block_hash=0x2a92ff72dad302d39fa80ef81522f0ccb27dc903255b618dfc4feddb22a8f80d l2_block_number=0 l1_block_hash=0xf39446e09aeca67452545d06a6e6a6a11184575ecf421f9306cf3602febf93ba l1_block_number=1 regolith_time="(not configured)" canyon_time="(not configured)" delta_time="(not configured)" ecotone_time="(not configured)" fjord_time="(not configured)" granite_time="(not configured)" holocene_time="(not configured)" isthmus_time="(not configured)" jovian_time="(not configured)" interop_time="(not configured)" cel2_time="(not configured)"
caff-node-1          | t=2025-06-05T11:50:25+0000 lvl=info msg="Initializing rollup node" version=v0.0.0
caff-node-1          | t=2025-06-05T11:50:25+0000 lvl=error msg="Error initializing the rollup node" err="failed to init L1: failed to validate the L1 config: incorrect L1 RPC chain id 1, expected 900"
caff-node-1          | t=2025-06-05T11:50:25+0000 lvl=crit msg="Application failed" message="failed to setup: unable to create the rollup node: failed to init L1: failed to validate the L1 config: incorrect L1 RPC chain id 1, expected 900"
caff-node-1 exited with code 1
```

shenkeyao commented:
~~Thanks @philippecamacho. Looks like it's due to the Geth version upgrade, which has a different requirement on the parameters in the genesis file. I'm working on a fix now.~~

Caff node errors fixed.

shenkeyao committed:
Fix chain ID

shenkeyao committed:
Add op-geth, fix hashes, remove l1 genesis file.

shenkeyao committed:
Fix op geth, update hashes

shenkeyao committed:
Fix caff node errors

philippecamacho commented:
I still have some errors, but they are different now.

```
> git log
commit e79fbf8d190d42fa3b236c992f1710d656826042 (HEAD -> keyao/docker-for-deploy, origin/keyao/docker-for-deploy)
Author: Keyao Shen <shenkeyao@gmail.com>
Date:   Sun Jun 8 22:16:50 2025 -0700

    Fix caff node errors
    
> docker compose down -v && docker volume prune -f
> docker compose up --build -d && docker compose logs -f | grep caff-node-1

caff-node-1 exited with code 1
caff-node-1          | t=2025-06-09T20:02:29+0000 lvl=info msg="Not opted in to ProtocolVersions signal loading, disabling ProtocolVersions contract now."
caff-node-1          | t=2025-06-09T20:02:29+0000 lvl=info msg="Using default bootnodes, none provided."
caff-node-1          | t=2025-06-09T20:02:29+0000 lvl=info msg="Rollup Config" l2_chain_id=1 l2_network="unknown L2" l1_chain_id=1337 l1_network="unknown L1" l2_start_time=1728358574 l2_block_hash=0x989d7c9b1642192d130f73d6bd1f80c0719ed3fbe15ea1219a87af85025ea0d1 l2_block_number=0 l1_block_hash=0x01f56292a859b15483339344a0e81cbd8357aa043eaf78e73d1a19208b6a00be l1_block_number=1 regolith_time="(not configured)" canyon_time="(not configured)" delta_time="(not configured)" ecotone_time="(not configured)" fjord_time="(not configured)" granite_time="(not configured)" holocene_time="(not configured)" isthmus_time="(not configured)" jovian_time="(not configured)" interop_time="(not configured)" alt_da="{DAChallengeAddress:0x0000000000000000000000000000000000000000 CommitmentType:GenericCommitment DAChallengeWindow:160 DAResolveWindow:160}" cel2_time="(not configured)"
caff-node-1          | t=2025-06-09T20:02:29+0000 lvl=info msg="Initializing rollup node" version=v0.0.0
caff-node-1          | t=2025-06-09T20:02:29+0000 lvl=error msg="Error initializing the rollup node" err="failed to init L1: failed to validate the L1 config: incorrect L1 genesis block hash 0x23256e053b7bc54589ed710887bc3b05accaefb59f3d702a997b8c289009425f, expected 0x01f56292a859b15483339344a0e81cbd8357aa043eaf78e73d1a19208b6a00be"
caff-node-1          | t=2025-06-09T20:02:29+0000 lvl=crit msg="Application failed" message="failed to setup: unable to create the rollup node: failed to init L1: failed to validate the L1 config: incorrect L1 genesis block hash 0x23256e053b7bc54589ed710887bc3b05accaefb59f3d702a997b8c289009425f, expected 0x01f56292a859b15483339344a0e81cbd8357aa043eaf78e73d1a19208b6a00be"
```

shenkeyao commented:
> docker compose down -v && docker volume prune -f

@philippecamacho, sorry, I should have mentioned this earlier--running the command with `-v` would regenerate the hashes, so we would get the hash mismatch error. I'll update the PR description with more info!

shenkeyao committed:
Commit jwt file as well

shenkeyao committed:
Remove jwt

shenkeyao commented:
@philippecamacho I added more details to the "How to test this PR" section in the PR description. In your case, could you try replacing the l1 hash in `rollup_devnet.json` with `0x23256e053b7bc54589ed710887bc3b05accaefb59f3d702a997b8c289009425f`, then retry the `docker compose up --build -d && docker compose logs -f | grep caff-node-1` command?

philippecamacho commented:
> @philippecamacho I added more details to the "How to test this PR" section in the PR description. In your case, could you try replacing the l1 hash in `rollup_devnet.json` with `0x23256e053b7bc54589ed710887bc3b05accaefb59f3d702a997b8c289009425f`, then retry the `` command?

I get this error after applying the change you suggest:
```
caff-node-1 exited with code 1
caff-node-1          | t=2025-06-09T22:27:43+0000 lvl=info msg="Not opted in to ProtocolVersions signal loading, disabling ProtocolVersions contract now."
caff-node-1          | t=2025-06-09T22:27:43+0000 lvl=info msg="Using default bootnodes, none provided."
caff-node-1          | t=2025-06-09T22:27:43+0000 lvl=info msg="Rollup Config" l2_chain_id=1 l2_network="unknown L2" l1_chain_id=1337 l1_network="unknown L1" l2_start_time=1728358574 l2_block_hash=0x989d7c9b1642192d130f73d6bd1f80c0719ed3fbe15ea1219a87af85025ea0d1 l2_block_number=0 l1_block_hash=0x23256e053b7bc54589ed710887bc3b05accaefb59f3d702a997b8c289009425f l1_block_number=1 regolith_time="(not configured)" canyon_time="(not configured)" delta_time="(not configured)" ecotone_time="(not configured)" fjord_time="(not configured)" granite_time="(not configured)" holocene_time="(not configured)" isthmus_time="(not configured)" jovian_time="(not configured)" interop_time="(not configured)" alt_da="{DAChallengeAddress:0x0000000000000000000000000000000000000000 CommitmentType:GenericCommitment DAChallengeWindow:160 DAResolveWindow:160}" cel2_time="(not configured)"
caff-node-1          | t=2025-06-09T22:27:43+0000 lvl=info msg="Initializing rollup node" version=v0.0.0
caff-node-1          | t=2025-06-09T22:27:43+0000 lvl=error msg="Error initializing the rollup node" err="failed to init L2: failed to get L2 genesis blockhash: failed to determine L2BlockRef of height 0, could not get payload: post-merge block header requires zeroed block nonce field, but got: \x00\x00\x00\x00\x00\x00\x00B"
caff-node-1          | t=2025-06-09T22:27:43+0000 lvl=crit msg="Application failed" message="failed to setup: unable to create the rollup node: failed to init L2: failed to get L2 genesis blockhash: failed to determine L2BlockRef of height 0, could not get payload: post-merge block header requires zeroed block nonce field, but got: \x00\x00\x00\x00\x00\x00\x00B"

```

shenkeyao commented:
> caff-node-1          | t=2025-06-09T22:27:43+0000 lvl=crit msg="Application failed" message="failed to setup: unable to create the rollup node: failed to init L2: failed to get L2 genesis blockhash: failed to determine L2BlockRef of height 0, could not get payload: post-merge block header requires zeroed block nonce field, but got: \x00\x00\x00\x00\x00\x00\x00B"

@philippecamacho I used to get this error too, but that should have been fixed by the new genesis file and some other changes. I think we'd need a mob programming session tomorrow, since I can't reproduce this error anymore. 🤔

shenkeyao committed:
Add L1 genesis file

shenkeyao commented:
New changes:
* Added a new L1 genesis file, the same as what’s used in the dev mode.
* Added a command to the L1 service to fetch the L1 genesis file.
* Renamed the L2 genesis file.

@philippecamacho, could you try the following?
* `docker run --rm   -v $(pwd)/../config:/config   -v espresso_op-geth-data:/data   us-docker.pkg.dev/oplabs-tools-artifacts/images/op-geth:v1.101503.2-rc.3   init --datadir=/data --state.scheme=path /config/l2-genesis-devnet.json` to relink the L2 genesis file due to the renaming.
* `docker compose down -v && docker volume prune -f` to clear volumes.
* `docker compose up caff-node` to see if there's any error related to inconsistent hash or genesis setting.

shenkeyao committed:
Add back l2 genesis file but rename it

philippecamacho commented:
@shenkeyao 

Yeah, looks like it is working:
```
leloup@leloup-Legion-7-16ITHg6:~/repositories/espressosys/optimism-espresso-integration/espresso$ docker compose up caff-node
WARN[0000] volume "espresso_op-geth-data" already exists but was not created by Docker Compose. Use `external: true` to use an existing volume 
[+] Running 7/7
 ✔ Network espresso_default                Created                                                                                                                                                                                                0.0s 
 ✔ Volume "espresso_l1-data"               Created                                                                                                                                                                                                0.0s 
 ✔ Volume "espresso_espresso-data"         Created                                                                                                                                                                                                0.0s 
 ✔ Container espresso-espresso-dev-node-1  Created                                                                                                                                                                                                0.2s 
 ✔ Container espresso-l1-1                 Created                                                                                                                                                                                                0.1s 
 ✔ Container espresso-op-geth-1            Created                                                                                                                                                                                                0.1s 
 ✔ Container espresso-caff-node-1          Created                                                                                                                                                                                                0.1s 
Attaching to caff-node-1
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="Not opted in to ProtocolVersions signal loading, disabling ProtocolVersions contract now."
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="Using default bootnodes, none provided."
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="Rollup Config" l2_chain_id=1 l2_network="unknown L2" l1_chain_id=1337 l1_network="unknown L1" l2_start_time=1728358574 l2_block_hash=0x989d7c9b1642192d130f73d6bd1f80c0719ed3fbe15ea1219a87af85025ea0d1 l2_block_number=0 l1_block_hash=0xfc3c494d08d1e07af2b32ab3a4b771cdb3de9272bfe48017d7049d6af7d7e555 l1_block_number=0 regolith_time="(not configured)" canyon_time="(not configured)" delta_time="(not configured)" ecotone_time="(not configured)" fjord_time="(not configured)" granite_time="(not configured)" holocene_time="(not configured)" isthmus_time="(not configured)" jovian_time="(not configured)" interop_time="(not configured)" alt_da="{DAChallengeAddress:0x0000000000000000000000000000000000000000 CommitmentType:GenericCommitment DAChallengeWindow:160 DAResolveWindow:160}" cel2_time="(not configured)"
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="Initializing rollup node" version=v0.0.0
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=debug msg="Espresso Streamer namespace:" !BADKEY=1 !BADKEY=<nil> LOG_ERROR="Normalized odd number of arguments by adding nil"
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="Espresso streamer initialized" namespace=1 "next hotshot block num"=1 "polling hotshot polling interval"=500ms "hotshot urls"=[http://espresso-dev-node:24000]
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="loaded new runtime config values!" p2p_seq_address=0x0000000000000000000000000000000000000000
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="started p2p host" addrs="[/ip4/127.0.0.1/tcp/9222 /ip4/172.20.0.5/tcp/9222]" peerID=16Uiu2HAmBuWgSF2YM7dwpUuC6qoFzpipBGcNXGDS97VYDuhWnG7X
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="New local node record" seq=1749640183473 id=6cb181f497bfe72b6fb3de50f5502c0ec3bbb2d4981e29b7817c8bbc06d0cde2 ip="invalid IP" udp=58003 tcp=9222
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="started discovery service" p2p=discv5 enr=enr:-JS4QE8qalNR6HFe6I_1pTrw5CJWqVXQbCE3nzyCtFPY9zmyUXc3EI9imp9gsqE1zlxluj4GNpYRGhG9fsxer62niqOGAZdern6xgmlkgnY0h29wc3RhY2uCAQCJc2VjcDI1NmsxoQL03L0fbcWcrouHBBSQuxMjizj2fTCU4bhhDvTCTqQ9boN0Y3CCJAaDdWRwguKT id=6cb181f497bfe72b6fb3de50f5502c0ec3bbb2d4981e29b7817c8bbc06d0cde2
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=debug msg="Creating RPC handler"
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="registered API" route="" namespace=optimism
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="registered API" route="" namespace=opp2p
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="P2P RPC enabled"
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="registered API" route="" namespace=admin
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="Admin RPC enabled"
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="Starting JSON-RPC server"
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="Started RPC server" endpoint=http://127.0.0.1:9545
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="Started JSON-RPC server" addr=127.0.0.1:9545
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="metrics disabled"
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="Sequencer config for finality" SequencerUseFinalized=false
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="Starting execution engine driver"
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="Starting driver" sequencerEnabled=false sequencerStopped=false recoverMode=false
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="Rollup node started"
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="State loop started"
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=debug msg="Sync process step"
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=warn msg="Deriver system is resetting" err="reset: cannot continue derivation until Engine has been reset"
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=debug msg="scheduling re-attempt with delay" attempts=1 delay=2.045804074s
caff-node-1  | t=2025-06-11T11:09:43+0000 lvl=info msg="L
```

shenkeyao committed:
Merge branch 'celo-integration-rebase-13' into keyao/docker-for-deploy

shenkeyao committed:
Update gitignore, remove unnecessary commands

philippecamacho commented:
@shenkeyao Could you please update the `README_ESPRESSO.md` file with the instructions to run the Docker file?

shenkeyao committed:
Update readme

shenkeyao commented:
> @shenkeyao Could you please update the `README_ESPRESSO.md` file with the instructions to run the Docker file?

@philippecamacho updated!

philippecamacho commented on README_ESPRESSO.md:
````diff
@@ -208,3 +208,60 @@ git submodule update --init --recursive
+
+* Build and start all services in the background.
+```
+docker compose up --build -d`
````
```suggestion
docker compose up --build -d
```

philippecamacho commented on README_ESPRESSO.md:
````diff
@@ -208,3 +208,60 @@ git submodule update --init --recursive
+
+* If the genesis file is updated, initialize the chain data directory with the updated file.
+```
+docker run --rm \
+  -v $(pwd)/../config:/config \
+  -v espresso_op-geth-data:/data \
+  us-docker.pkg.dev/oplabs-tools-artifacts/images/op-geth:v1.101503.2-rc.3 \
+  init --datadir=/data --state.scheme=path /config/<genesis-file-name>.json
````
```suggestion
docker run --rm \
  -v $(pwd)/../config:/config \
  -v espresso_op-geth-data:/data \
  us-docker.pkg.dev/oplabs-tools-artifacts/images/op-geth:v1.101503.2-rc.3 \
  init --datadir=/data --state.scheme=path /config/l2-genesis-devnet.json
```

philippecamacho replied:
I think it is better to put the name of the file in this case.

shenkeyao replied:
I added a note for the file name rather than using the L2 genesis file directly, because it might be the L1 genesis file.

Also added a reminder in `docker-compose.yml` to update the op-geth version.

shenkeyao committed:
Update readme

philippecamacho reviewed (approved):
LGTM
